channels/ON-CHAIN.md

```diff
@@ -60,18 +65,27 @@ Channel(cid).amount := initiator_amount + responder_amount
 
 Serialization defined [here](../serializations.md#channel-create-transaction)
 
-- `initiator`: public key/address of the initiating peer
-- `responder`: public key/address of the responding peer
-- `initiator_amount`: unsigned amount of tokens the initiator commits to the channel
-- `responder_amount`: unsigned amount of tokens the responder commits to the channel
-- `lock_period`: minimal block height interval between a channel_close_solo/last channel_slash transaction and the channel_settle transaction
-- `ttl`
-- `fee`
-- `delegates`: a list of delegate addresses. The delegates play a role in the
-  solo closing sequence: except for the participants of the channel, they are
-  the only ones that can provide a slash transaction.
-- `state_hash`: the root hash of the channel state tree; This is not validated, just kept in the channel's object
-- `nonce`
+- `initiator_id`: account id of the initiating peer
+- `responder_id`: account id of the responding peer
+- `initiator_amount`: unsigned amount of coins the initiator commits to the
+  channel
+- `responder_amount`: unsigned amount of coins the responder commits to the
+  channel
```

Comment:
@knarz (Potentially in separate new PR) Could you please detail semantics of `channel_reserve` in `/channels/ON-CHAIN.md#channel_withdraw` ?

I see the requirements below mention `channel_reserve` but that is not listed/described among fields here. This may surprise reader. (It is present in the linked serializations.md ) I tried to understand semantics of `channel_reserve` from the rest of the document, and I would have expected it to be mentioned in `channel_withdraw` though I did not find it there. Hence my request.

@velzevur @uwiger
* The withdraw tx [checks the double of the reserve against one combined amount ("channel amount" - not sure of its semantics)](https://github.com/aeternity/epoch/blob/f0493f6eb0e2fec4fabe639da944e7153f4ebb89/apps/aechannel/src/aesc_withdraw_tx.erl#L291-L292), rather than checking the reserve against each single party (initiator and responder) as the create tx does. Is this intended?
* The force progress tx [does not check that each single party (initiator and responder) has at least channel reserve in its amount](https://github.com/aeternity/epoch/blob/f0493f6eb0e2fec4fabe639da944e7153f4ebb89/apps/aechannel/src/aesc_utils.erl#L522). Is this intended?
  * If introducing such check we may consider also to rather remove such unspendable reserve from each account (if it is an unspendable amount per party) *before* running the contract and re-adding it afterwards - so to prevent contract to cause miner to spend computational effort with those coins while it should not have done so.

Reply:
#### Withdrawal tx
Since in the withdraw transaction it is only the `state_hash` of the new off-chain state tree that is being provided, we can not enforce that both participants have at least `channel_reserve` tokens left in their off-chain balances. Thus we can not enforce this further than the on-chain total channel amount never falls lower than the double `channel_reserve` amount.
Alternative would be providing a proof of inclusion for both account's balances (or a full set of trees) in the withdrawal transaction. This way we could validate amounts.
I'd rather go with the current _optimistic check_ approach

#### Force progress
Force progress includes a shift of tokens from the caller to the off-chain contract. A force progress transaction then only potentially lowers the amount of tokens possesed by the `caller` (forcer of progress) and should not validate the other participant. The `aesc_utils:process_force_progress/7` indeed does check the [caller of the contract doesn't spend too much](https://github.com/aeternity/epoch/blob/f0493f6eb0e2fec4fabe639da944e7153f4ebb89/apps/aechannel/src/aesc_offchain_update.erl#L268-L273) but this is burried deep in the channel internals.

Reply:
> indeed does check the caller of the contract doesn't spend too much

I am not sure about the following case though did not arrive at checking. If:
* Parties start at A=20 and B=40, and reserve is 10 (hence 2*10).
* There is withdrawal of 40 from B -> valid (I am ignoring fees) because leaving 20 that is > 2*10.
* Then there is force progress that removes 10 from A -> valid because A (was 20 and) becomes 10. BUT the total amount is less than 2*10 (10 + 0).

Reply:
yes, this is possible, indeed

Reply:
We're going to address the `channel_reserve` in a separate PR